### PR TITLE
Getting the docs building on ReadTheDocs

### LINKED ISSUE #33: Move training material to own repository
As suggested by @dalonsoa on Teams.

The idea is that the training materials, while very good, are essentially a separate
project from the virtual rainforest code and probably would be best placed in their own
repository.

I happen to agree with @dalonsoa about this and almost didn't raise this as an issue
because this suggestion didn't meet with universal approval, however this is exactly
what happens with real issues! So I think it makes sense to have this listed as an
issue. You guys can discuss it on here and eventually you can make a decision either to
act on it or to close it.

### LINKED ISSUE #49: Documentation fails to build complaining about vr-python3 not found
**Describe the bug**
Try to build the documentation and it fails with:

```bash
Exception occurred:
  File "/Users/dalonsoa/Documents/Projects/virtual_rainforest/.venv/lib/python3.9/site-packages/jupyter_client/kernelspec.py", line 294, in get_kernel_spec
    raise NoSuchKernel(kernel_name)
jupyter_client.kernelspec.NoSuchKernel: No such kernel named vr_python3
```

**To Reproduce**

Assuming you have the virtual environment activate, just run:

```bash
cd docs
make clean
make html
```

**Expected behavior**
The documentation should build without issues.

**Screenshots**
Not applicable

**Desktop (please complete the following information):**

- OS: [e.g. iOS] MacOS
- Version [e.g. 22] 12.4

**Additional context**
Using the plain develop branch with last commit dab354486

### LINKED ISSUE #74: Get the website building and discuss structure
**Is your feature request related to a problem? Please describe.**
As it currently exists, the package documentation doesn't build anywhere, and doesn't have a clear structure.

**Describe the solution you'd like**
Would be good to setup a system to build the docs as a site, and in doing that we would have to agree upon a structure for the docs folders. We could probably build the site using [Read the Docs](https://readthedocs.org)

**Describe alternatives you've considered**
We don't necessarily need to do this soon, we can stick with fragmentary documentation for the time being, but we are definitely going to have to do this at some point.

### PR DESCRIPTION
**Pull request by davidorme**
# Description

This PR:

* Provides a `.readthedocs.yaml` configuration that installs the package via `poetry` on RTD.
* Updates to using `sphinx 5.3.0` (previously pinned under 4.0.0).
* Removes the training materials that had been put in the repo (Fixes #33).
* Sets up RTD to use the standard kernel name for doc building in `myst-nb` (Closes #49).
* Adds first demonstration of automatic documentation build on the API.
* Fixes some markdown issues and docstring style issues.
* The documentation now builds on RTD (Closes #74). 

The documentation structure could still do with work - but that is a longer term thing.

This PR includes changes from the `feature/grid-revamp` branch which is PR #84, so that should be pulled in first.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [x] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: docs/source/virtual_rainforest/soil/soil_details.md [hunk 1]
```diff
@@ -111,30 +111,30 @@ significant nitrogen cycle processes that involve inorganic forms of nitrogen. F
 reason a number of inorganic nitrogen pools are additionally defined. They are as
 follows:
 
-### Combined ammonia ($\\ce{NH\_{3}}$) and ammonium ($\\ce{NH\_{4}^{+}}$) pool
+### Combined ammonia ($\ce{NH_{3}}$) and ammonium ($\ce{NH_{4}^{+}}$) pool
```
Comment (jacobcook1995) [inline docs/source/virtual_rainforest/soil/soil_details.md:114]: Are these rendering right now? The [read the docs site](https://virtual-rainforest.readthedocs.io/en/feature-getting_docs_working/virtual_rainforest/soil/soil_details.html) still shows them as badly formatted, but I can't tell if the site is showing a previous version

**Review by jacobcook1995 — COMMENTED** [review]
Aside my query about the chemical symbol notation everything looks to me, though my knowledge of online documentation systems is very limited

File: docs/source/virtual_rainforest/soil/soil_details.md (hunk 1 above)
Comment (dalonsoa) [reply docs/source/virtual_rainforest/soil/soil_details.md:114]: They look great, now! You can see the revision rendered at the bottom of the page.

**Review by dalonsoa — APPROVED** [review]
I'm not sure if I should be looking at any specific aspect, but for what I can tell after skimming through Read The Docs, it all seems to format well and looking appealing.

**Pull request merged**